dsgrid/dataset/dataset_schema_handler_one_table.py

```diff
@@ -167,7 +228,48 @@ def make_project_dataframe(
             ld_df = self._apply_fraction(ld_df, {VALUE_COLUMN}, mapping_manager)
             project_metric_records = self._get_project_metric_records(project_config)
             ld_df = self._convert_units(ld_df, project_metric_records, mapping_manager)
+            input_dataset = project_config.get_dataset(self._config.model.dataset_id)
             ld_df = self._convert_time_dimension(
-                ld_df, project_config, VALUE_COLUMN, mapping_manager
+                load_data_df=ld_df,
+                to_time_dim=project_config.get_base_time_dimension(),
+                value_column=VALUE_COLUMN,
+                mapping_manager=mapping_manager,
+                wrap_time_allowed=input_dataset.wrap_time_allowed,
+                time_based_data_adjustment=input_dataset.time_based_data_adjustment,
+                to_geo_dim=project_config.get_base_dimension(DimensionType.GEOGRAPHY),
             )
             return self._finalize_table(context, ld_df, project_config)
+
+    def make_mapped_dataframe(
+        self,
+        context: QueryContext,
+        geography_dimension: DimensionBaseConfigWithFiles | None = None,
+        metric_dimension: DimensionBaseConfigWithFiles | None = None,
+        time_dimension: TimeDimensionBaseConfig | None = None,
```

Comment:
Can the user not map sector, subsector, end use, model_year, scenario? Why these three specific dimension types?

Reply:
They can. We have special handling for these three. We may add time zone through geography. We convert units through metric. Time is its own thing. The others don't have special handling.